### DISCUSSION TITLE
Combining gas phase fractions; RT correction

### DISCUSSION THREAD

**Discussion by KristianSwearingen** [vdemichev/DiaNN#655] [Q&A]
I collected DIA data using gas phase fractionation: each sample was analyzed by six LC-MS runs with identical instrument parameters, except that run one is 398-502 m/z, run two is 498-602 m/z...run six is 898-1002 m/z.
How can I combine the six gas phase fractions for each sample so that they are treated as a single sample rather than being treated as six individual samples?
Is there a way to correct for retention time drift that happens over the course of the experiment?

**Comment by tvpham** [comment]
I have recently analyzed a dataset where each sample consists of 3 fractions (not the fractionation like yours). Each raw file name is a combination of sample and fraction A1 A2 A3 B1 B2 B3 and so on. You can then just do a normal DIA-NN search and process the long format output report.tsv as follows.

- use R (or any other tool such as awk, sed if you like) to split the column  'File Name' to sample and fraction.
- make a new tab-separated file with the additional two new columns.
- call process_long_format from the iq package (disclaimer: I am author of iq), treating fraction as additional secondary_id so that ions in the same fraction will be compared against each other.

The R code is something like this:

```
raw <- read.delim("report.tsv")

f <- gsub("^[A-Z]*", "", raw$File.Name) # remove A-Z to get fraction
s <- gsub(".$", "", raw$File.Name) # remove last character to get sample name

raw2 <- cbind(raw, s, f)

write.table(raw2, file = "new_data.txt", sep = "\t", row.names = FALSE, quote = FALSE)

require(iq)

process_long_format("new_data.txt", output_filename = "protein_report.txt", 
                    sample_id = "s", 
                    second_id = c("Precursor.Id", "f"),
                    primary_id = "Protein.Group",  annotation_col = c("Protein.Names", "Genes"), 
                    filter_double_less = c("Global.Q.Value" = "0.01", "Global.PQ.Q.Value" = "0.01")) 

```

I am curious if this works for you.

Thang

**Reply by KristianSwearingen** [reply]
Thanks for your suggestion. Being honest, this approach is above my skill set at the moment. I am accustomed to working within a GUI using the built-in features of a program, but I do not have the resources to implement additional analysis in R.

By the fact that you took this approach for your data, does this mean that DIA-NN does not have the built-in capability to combine individual fractions into a single sample for the purposes of peptide identification and quantification?

**Comment by LeeSCantrell** [comment]
In addition to what @tvpham suggested, do make sure you control for qvalue... merging libraries or independently evaluated qvalues may result in improper assumption. For a no hassle evaluation, I suggest library free + mbr with filtering done by library q value. If you aren't going that route, in-house protein inference should be done and you should probably set a filter at 1/n_fractions % q-value to avoid incorrect assumptions of independence.

**Reply by KristianSwearingen** [reply]
Thanks. This is part of why I don't want to analyze the fractions independently and then combine them after the fact. I'd like for the individual gas-phase fraction runs to be combined and treated as a single sample. 
I'm used to working in EncyclopeDIA. In that program, I can merge the data files into a single file prior to analysis, but there is no RT alignment prior to merging, so any RT drift over the course of the acquisition is not corrected. I can analyze each sample individually and the RT alignment works well, but EncyclopeDIA cannot merge the individual results files into a single result file to be evaluated and quantified. That's what I'm trying to figure out if DIA-NN can do.

**Reply by LeeSCantrell** [reply]
Hi Kristian,  I generally agree. 

A few quick points on that EncyclopeDIA merge. As you note, it doesn't do RT correction or produce mzML files for search in other pipelines. I think more importantly, if you are working with overlapping/stagger window formats, if there is a partition in your gpf where you only measure 1x per cycle, it will drop that search space. For example, if fraction 1 is collected from 400-500 and 402-498, and subsequent fractions in similar structure, the m/z range from 498-500 will be dropped. This feature is then more useful for quantitative experiments with non-overlapping windows when percolator should only be run 1x for protein level inference.

I have been curious about whether diann fdr is suited for GPF data in the first place. It can perform local fdr, but I don't believe it necessarily has global protein/gene groups level fdr calculated in a robust manner to handle the oddities of these experiments. E.g., protein profile measured in the first fraction is not necessarily consistent with other fractions. 
I'd also be careful to evaluate precursor intensity as only from the unnormalized value and not normalized values since the sliding median intensity in rt makes assumptions of heterogeneity that aren't necessarily valid in quant gpf experiments. Its possible this effect is minimized by maxLFQ, but I'm not certain.

**Reply by vdemichev** [reply]
Hi Kristian, sorry, I guess I somehow missed this thread. As I understand, each biological sample was acquired in multiple GPF fractions? That is GPF fractions need to be used for quantitation? There's unfortunately no way to do this fully automatically in DIA-NN, i.e. need to use R (very minimal). 

What I would suggest:
- Analyse all runs in one go with MBR.
- Open the report in R, filter Lib.PG.Q.Value <= 0.01, and introduce a column which is a sample ID (i.e. the same for all GPF runs of a sample).
- Quantify proteins on per-sample basis using the 'iq' R package. Might need to remove duplicate precursor IDs, e.g. df <- df[!duplicated(paste0(df$Sample.Id,df$Precursor.Id)),].
- Optionally (but this will likely be beneficial): perform protein-level normalisation, I think there are quite a number of packages for doing this.

Hi Lee, all types of q-values in DIA-NN, including all global ones, are calculated based on target-decoy competion, so the experiment design should have little to no effect on their validity. 

Best,
Vadim

**Comment by KristianSwearingen** [comment]
Thanks Vadim for your suggestions.
To answer your questions: yes, each biological replicate was acquired in multiple GPF fractions. That is, one sample in one vial was injected six different times. The goal is to combine the six individual search results into a single result.

Based on your and Thang's suggestion, it sounds like you are saying I can combine the peptide-level quants in iq and then have the protein-level quant inferred from that. I'll give it a try.
